### PR TITLE
Remove shorthand flags for "mount", "pretty", and "no-resolve"

### LINKED ISSUE #24433: [1.12] evaluate new single-letter flags before GA
With the SwarmKit integration, many new commands and options were added; we should be careful when adding shorthand (single-letter) flags to commands, and only use shorthand flags for frequently used options.

In doubt; remove the shorthand flag, so that we can decide later when really needed.

Opening this issue so that we don't forget to evaluate all new flags before GA

### PR DESCRIPTION
**Pull request by thaJeztah**
fixes https://github.com/docker/docker/issues/24433

### CODE REVIEW

**Comment by vdemeester** [comment]
@thaJeztah on `--pretty` I'm even for removing the flag and implement a `--format pretty` format. I have almost something working locally :wink:

**Comment by thaJeztah** [comment]
@vdemeester yes, that's an alternative :-) I opened this PR to get things moving, but ill update on whatever we decide

**Comment by tiborvass** [comment]
+1 for this for now, the --pretty question is an orthogonal one. People disagreed with `--format pretty` since it would complicate the help output and would have similar problems to `--filter`.
On the other hand, `--mount` has that problem too so I'm not convinced.

Either way, LGTM for this as-is.

**Comment by thaJeztah** [comment]
oh, ping @albers @sdurrheimer if you could have a quick look if I didn't mess up the completion changes

**Comment by icecrime** [comment]
LGTM as it is, thanks!

**Comment by vdemeester** [comment]
all right, LGTM as is too 🐮

**Comment by albers** [comment]
bash completion LGTM. :heart:

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -808,7 +808,7 @@ __docker_node_subcommand() {
             _arguments $(__docker_arguments) \
                 $opts_help \
                 "($help -f --format)"{-f=,--format=}"[Format the output using the given go template]:template: " \
-                "($help -p --pretty)"{-p,--pretty}"[Print the information in a human friendly format]" \
+                "($help --pretty)--pretty[Print the information in a human friendly format]" \
```
Comment (vdemeester) [inline contrib/completion/zsh/_docker]: I think you can simply do `"($help)--pretty[…]" \`

File: contrib/completion/zsh/_docker [hunk 2]
```diff
@@ -1072,7 +1072,7 @@ __docker_service_subcommand() {
         "($help)--limit-cpu=[Limit CPUs]:value: "
         "($help)--limit-memory=[Limit Memory]:value: "
         "($help)--mode=[Limit Memory]:mode:(global replicated)"
-        "($help)*"{-m=,--mount=}"[Attach a mount to the service]:mount: "
+        "($help)*--mount=[Attach a mount to the service]:mount: "
```
Comment (vdemeester) [inline contrib/completion/zsh/_docker]: I think the `*` is not required here (and below)

**Comment by tiborvass** [comment]
@vdemeester i'm so sorry i got github-cached

**Comment by vdemeester** [comment]
@tiborvass np, I'll make a follow-up PR to fix :wink:

File: contrib/completion/zsh/_docker (hunk 2 above)
Comment (vdemeester) [inline contrib/completion/zsh/_docker]: nvm on this one :sweat_smile:

**Comment by thaJeztah** [comment]
Thanks @vdemeester

**Comment by aluzzardi** [comment]
@vdemeester 
> on --pretty I'm even for removing the flag and implement a --format pretty format. 

Yeah, I agree with that. I would probably call it `--output` (`-o pretty` / `-o json`) since `--format` is already taken.

Also, I would make pretty the default if stdin is a tty (e.g. it's probably a human typing the command).

Thoughts?

/cc @tiborvass @icecrime

**Comment by tiborvass** [comment]
@aluzzardi tty-based discrimination is a bad idea. You use it in the terminal and you expect things to be some way, and then you put it in a script with no tty and it completely breaks.

I'm -1 on `--output`. `--pretty` or `--format pretty`, although some people had some concerns about overloading --format which today only accepts Go template strings.

**Comment by aluzzardi** [comment]
@tiborvass There are 0 chances I'm trying to parse json with my eyes :)

The fact that docker inspect prints json is a huge design mistake made years ago.
It's kinda crazy that there's no way to "see" a container without getting a blob of json.

**Comment by tiborvass** [comment]
@aluzzardi I know and I agree. We can't change it now.

**Comment by vdemeester** [comment]
@tiborvass @aluzzardi I still like the idea of `--format pretty` 👼. `--format` *already support* specific formats (`table`, `raw`) so I feel `pretty` could be another one 👼.

That beeing said, we're way to close to GA to change `--pretty` or `--format` now :stuck_out_tongue_closed_eyes:

**Pull request merged**